crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -428,7 +428,33 @@ where
         &self,
         number: BlockNumber,
     ) -> ProviderResult<Option<StoredBlockBodyIndices>> {
-        self.database.block_body_indices(number)
+        if let Some(indices) = self.database.block_body_indices(number)? {
+            Ok(Some(indices))
+        } else if let Some(state) = self.canonical_in_memory_state.state_by_number(number) {
+            // we have to construct the stored indices for the in memory blocks
+            //
+            // To calculate this we will fetch the anchor block and walk forward from all parents
+            let mut parent_chain = state.parent_state_chain();
+            parent_chain.reverse();
+            let anchor_num = state.anchor().number;
+            let mut stored_indices = self
```

Comment:
If I'm reading this correctly, then we need to do a + 1 here on the `first_tx_num`

https://github.com/paradigmxyz/reth/blob/48d4c798f9a96aa50dfe11b9f45212e33669aec4/crates/stages/stages/src/stages/bodies.rs#L127-L128